src/npm.ts

```diff
@@ -0,0 +1,29 @@
+import { Job } from "@brigadecore/brigadier";
+
+export const npmReleaseImg = "brigadecore/npm-release:edge";
+
+export class NPMReleaseJob extends Job {
+  constructor(name: string, image?: string) {
+    if (image == undefined) {
+      image = npmReleaseImg;
+    }
+    super(name, image);
+
+    let checkEnv = '\
+      if [ -z "$NPM_TOKEN" ]; then \
+        echo "NPM_TOKEN for npm release must be provided via job environment" && exit 1 ; \
+      fi ; \
+      if [ -z "$VERSION" ]; then \
+        echo "VERSION for npm release must be provided via job environment" && exit 1 ; \
+      fi ; '
+
+    this.tasks = [
+      checkEnv,
+      "echo '//registry.npmjs.org/:_authToken=${NPM_TOKEN}' > .npmrc",
+      "cat package.json | jq '.version |= \"${VERSION}\"' > package.json.tmp",
+      "rm package.json",
+      "mv package.json.tmp package.json",
+      "npm publish"
```

Comment:
I don't know exactly the behaviour of `npm publish` if the package fails to build - this is irrelevant in how _we_ use this job, since we always build _and test_ before publishing, but it might be worth investigating and adding a note in the readme for users.

Reply:
Ah good question.  Indeed, a mention in the README, perhaps around what logs/output to look for in the event of a publishing failure, may be in order.